sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -396,6 +447,9 @@ def _getAssociatedImage(self, imageKey):
 
     @methodcache()
     def getTile(self, x, y, z, pilImageAllowed=False, numpyAllowed=False, **kwargs):
+        if self._levels is None:
```

Comment:
I don't see anywhere where `self._levels` is set to None.  Maybe this needs to be done on `addTile`?

Reply:
This was previously done in `addTile`, but I recently changed it so that `addTile` sets `self._levels` to a valid value. I think I should take away this check and call to `_validateZarr`, since it was meant to fill those values when `addTile` didn't.

Reply:
You are setting `self.levels` in `addTile`, not `self._levels`.  `self._levels` is a list of lists indicating which dataset is used for different resolutions (clearly a longer, more descriptive variable name would have been better).

As a simple test to add, after calling `addTile` with some data, call `getTile(0, 0, 0)` and see that it returns something.

Reply:
Oh I see the distinction now. I can put the line back in `addTile` where we set `self._levels = None`.